app/javascript/packs/suspendUserModal.js

```diff
@@ -0,0 +1,103 @@
+import { showWindowModal } from '@utilities/showModal';
+import { request } from '@utilities/http';
+
+const suspendUser = async ({ event, userId, username, suspensionReason }) => {
+  event.preventDefault();
+  closeSuspendUserModal();
+
+  try {
+    const response = await request(
+      `/admin/member_manager/users/${userId}/user_status`,
+      {
+        method: 'PATCH',
+        body: JSON.stringify({
+          id: userId,
+          user: {
+            note_for_current_role: suspensionReason,
+            user_status: 'Suspend',
+          },
+        }),
+        credentials: 'same-origin',
+      },
+    );
+
+    const outcome = await response.json();
+
+    /* eslint-disable no-restricted-globals */
+    if (outcome.success) {
+      top.addSnackbarItem({
+        message: `Success: "${username}" has been suspended.`,
+        addCloseButton: true,
+      });
+    } else {
+      top.addSnackbarItem({
+        message: `Error: something went wrong; ${username} NOT suspended.`,
+        addCloseButton: true,
+      });
+    }
+  } catch (error) {
+    top.addSnackbarItem({
+      message: `Error: ${error}`,
+      addCloseButton: true,
+    });
+  }
+};
+
+function closeSuspendUserModal() {
+  window.parent.document
+    .querySelector('#window-modal .crayons-modal__dismiss')
+    .click();
+}
+
+const modalContents = new Map();
+
+function getModalContents(modalContentSelector) {
+  if (!modalContents.has(modalContentSelector)) {
+    const modalContentElement =
+      window.parent.document.querySelector(modalContentSelector);
+    const modalContent = modalContentElement.innerHTML;
+
+    modalContentElement.remove();
+    modalContents.set(modalContentSelector, modalContent);
+  }
+
+  return modalContents.get(modalContentSelector);
+}
+
+function checkSuspensionReason(event) {
+  const { userId, username, suspensionReasonSelector } = event.target.dataset;
+  const suspendUserModal =
+    window.parent.document.getElementById('window-modal');
```

Comment:
```suggestion
  const suspendUserModal =
    window.parent.document.getElementById(WINDOW_MODAL_ID);
```
(suggestion) it might be nice to use the constant exported from `showModal`